tests/test_charm_builder.py

```diff
@@ -1155,9 +1165,11 @@ def mock_build_charm(self):
     fake_argv += ["-r" "reqs1.txt", "--requirement", "reqs2.txt"]
     with patch.object(sys, "argv", fake_argv):
         with patch("charmcraft.charm_builder.CharmBuilder.build_charm", new=mock_build_charm):
-            with pytest.raises(SystemExit) as raised:
-                charm_builder.main()
+            with patch("charmcraft.charm_builder.collect_charmlib_pydeps") as mock_collect_pydeps:
```

Comment:
out of curiosity: have you considered using the `mocker` fixture instead of "accumulating" context managers? Might make it easier to read.